Source/CombatExtended/CombatExtended/AmmoInjector.cs

```diff
@@ -1,325 +1,332 @@
-                            {
-                                Bill bill = billGiver.BillStack[i];
-                                if (!benchDef.AllRecipes.Exists(r => bill.recipe == r))
-                                {
-                                    billGiver.BillStack.Delete(bill);
-                                }
-                            }
-                        }
-                    }
-                }
-            }
-            benchDef.allRecipesCached = null;  // Set ammoCraftingStation.AllRecipes to null so it will reset
-        }
-
-        public static bool gunRecipesShowCaliber = false;
-        public static void AddRemoveCaliberFromGunRecipes()
-        {
-            var shouldHaveLabels = (Controller.settings.EnableAmmoSystem && Controller.settings.ShowCaliberOnGuns);
-
-            if (gunRecipesShowCaliber != shouldHaveLabels)
-            {
-                CE_Utility.allWeaponDefs.ForEach(x =>
-                {
-                    var ammoSet = x.GetCompProperties<CompProperties_AmmoUser>()?.ammoSet;
-
-                    if (ammoSet != null)
-                    {
-                        RecipeDef recipeDef = DefDatabase<RecipeDef>.GetNamed("Make" + x.defName, false);
-
-                        if (recipeDef != null)
-                        {
-                            var label = x.label + (shouldHaveLabels ? " (" + (string)ammoSet.LabelCap + ")" : "");
-
-                            recipeDef.UpdateLabel("RecipeMake".Translate(label));           //Just setting recipeDef.label doesn't update Jobs nor existing recipeUsers. We need UpdateLabel.
-                            recipeDef.jobString = "RecipeMakeJobString".Translate(label);   //The jobString should also be updated to reflect the name change.
-                        }
-                    }
-                });
-
-                gunRecipesShowCaliber = shouldHaveLabels;
-            }
-        }
-    }
-}
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Reflection;
+using RimWorld;
+using Verse;
+using UnityEngine;
+
+namespace CombatExtended
+{
+    /* Ammo Injection Handler
+     * 
+     * Automatically enables all ammo types defined and in use by at least one gun somewhere. Also enables appropriate crafting recipes and trading tags.
+     * This way we can have all ammo defs in one core mod and selectively enable the ones we need based on which gun mods are installed, avoiding duplication
+     * issues where multiple gun mods are adding the same ammo.
+     * 
+     * When ammo system is disabled automatically disables crafting and spawning of all ammo. 
+     * 
+     * Call Inject() on game start and whenever ammo system setting is changed.
+     */
+    internal static class AmmoInjector
+    {
+        public static readonly FieldInfo _allRecipesCached = typeof(ThingDef).GetField("allRecipesCached", BindingFlags.Instance | BindingFlags.NonPublic);

```

Comment:
Shouldn't need the FieldInfo since you aren't using it.  As an aside, if you had made that `private static readonly`, the compiler would have warned you it was unused.